### PR TITLE
Use updated DynFlags after CPP phase. Fixes #25

### LINKED ISSUE #25: CPP preprocessing
I had taken your `Preprocess` module and modified it a little to suit my needs for the [Argon](https://github.com/rubik/argon) project. I had encountered a bug in CPP preprocessing, and I see now that it's mentioned in ghc-exactprint's readme.

This is the commit in which I fixed the problem:
https://github.com/rubik/argon/commit/35ad16fab24e4be92948818ec137f48d1c3235ec

I annotated the commit diff to show exactly what changed. I could submit a PR if you want, but I'd need tests in order to verify my work.

### PR DESCRIPTION
**Pull request by rubik**
The change passes all the tests. Since they were probably passing before as well, you probably need more test cases for CPP. Tested with:

    $ stack init
    $ stack test

Note: Stack needs a `stack.yaml` file which is not present in the repository. I had to run `stack init` to create it. If you want I can submit a PR to add `stack.yaml` to the repo.

### CODE REVIEW

File: src/Language/Haskell/GHC/ExactPrint/Preprocess.hs [hunk 1]
```diff
@@ -185,7 +189,8 @@ getPreprocessedSrcDirectPrim cppOptions src_fn = do
   hsc_env <- GHC.getSession
   let dfs = GHC.extractDynFlags hsc_env
       new_env = GHC.replaceDynFlags hsc_env (injectCppOptions cppOptions dfs)
-  (dflags', hspp_fn) <- GHC.liftIO $ GHC.preprocess new_env (src_fn, Nothing)
+  (dflags', hspp_fn) <-
+      GHC.liftIO $ GHC.preprocess new_env (src_fn, Just (GHC.Cpp GHC.HsSrcFile))
```
Comment (mpickering) [inline src/Language/Haskell/GHC/ExactPrint/Preprocess.hs]: Is this change relevant?

Comment (rubik) [inline src/Language/Haskell/GHC/ExactPrint/Preprocess.hs]: Unfortunately I don't know exactly. Without a test covering this case it's difficult to tell. I deduced that it was necessary from GHC's "documentation".

**Comment by alanz** [comment]
Can you explain what problem this change is trying to solve? Perhaps by referring to a failing test case in another project.

File: src/Language/Haskell/GHC/ExactPrint/Preprocess.hs (hunk 1 above)
Comment (rubik) [inline src/Language/Haskell/GHC/ExactPrint/Preprocess.hs]: FWIW I replaced `Just` with `Nothing` in Argon's source code and tests passed. However, if we are doing CPP, why not using the right flag?

**Comment by mpickering** [comment]
https://github.com/alanz/ghc-exactprint/issues/25

File: src/Language/Haskell/GHC/ExactPrint/Preprocess.hs (hunk 1 above)
Comment (mpickering) [inline src/Language/Haskell/GHC/ExactPrint/Preprocess.hs]: There is no argument against using the right flag. I was wondering why this change caused the issue to be fixed.

Comment (mpickering) [inline src/Language/Haskell/GHC/ExactPrint/Preprocess.hs]: In fact, it is not clear that this is the right flag - I think we want to run the whole preprocessor pipeline rather than starting from the CPP phase. It is true that 99% of preprocessing is by CPP but other preprocessing might also be necessary.

Comment (rubik) [inline src/Language/Haskell/GHC/ExactPrint/Preprocess.hs]: Well yes, this is a marginal issue. The actual fix is to use the updated DynFlags.

**Comment by mpickering** [comment]
Thanks! I am going to revert the change to the phase flag but the rest looks good!
